wemake_python_styleguide/visitors/ast/annotations.py

```diff
@@ -124,23 +126,46 @@ def visit_arg(self, node: ast.arg) -> None:
         Raises:
             MultilineFunctionAnnotationViolation
             LiteralNoneAnnotation
+            UnionNoneViolation
 
         """
         self._check_arg_annotation(node)
         self.generic_visit(node)
 
+    def _check_slice_arg_for_none(self, slice_arg) -> bool:
+        if isinstance(slice_arg, ast.NameConstant):
+            return slice_arg.value is None
+        return False
+
     def _check_for_literal_none(self, node: ast.Subscript) -> None:
         annotation_name = cast(ast.Name, node.value)
         slice_index = cast(ast.Index, node.slice)
 
         if annotation_name.id != 'Literal':
             return
 
-        if not isinstance(slice_index.value, ast.NameConstant):
+        if self._check_slice_arg_for_none(slice_index.value):
+            self.add_violation(LiteralNoneViolation(node))
+
+    def _check_for_union_none(self, node: ast.Subscript) -> None:
+        annotation_name = cast(ast.Name, node.value)
```

Comment:
Can you please remove `cast`? It is not recommended use `cast`.

Reply:
If I remove `cast` expressions, I'll have some problems with mypy.